### PR TITLE
Create AddPointsCommand

### LINKED ISSUE #1: As a user learning to use the app, I can manually add points to a member
... so that I can reward them for their patronage

### PR DESCRIPTION
**Pull request by kinjalagarwal1810**
Implemented AddPointsCommand and all the related classes & test classes for v1.2
close #1

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/AddPointsCommandParser.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        } catch (ParseException pe) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddPointsCommand.MESSAGE_USAGE), pe);
+        }
+
+        Points points;
+        try {
+            points = ParserUtil.parsePoints(argMultimap.getValue(PREFIX_POINTS).get());
+        } catch (ParseException pe) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddPointsCommand.MESSAGE_USAGE), pe);
+        }
```
Comment (kokerinks) [inline src/main/java/seedu/address/logic/parser/AddPointsCommandParser.java:38]: I think the `ParseException` thrown by `parseName` and `parsePoints` don't have to be caught by try-catches(?)

I was referring to the scripts from e.g. `AddCommandParser.java` and this was how they handled getting values from `argMultimap`:

```
public AddCommand parse(String args) throws ParseException {
        ArgumentMultimap argMultimap =
                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ADDRESS, PREFIX_TAG);

        if (!arePrefixesPresent(argMultimap, PREFIX_NAME, PREFIX_ADDRESS, PREFIX_PHONE, PREFIX_EMAIL)
                || !argMultimap.getPreamble().isEmpty()) {
            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddCommand.MESSAGE_USAGE));
        }

        argMultimap.verifyNoDuplicatePrefixesFor(PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ADDRESS);
        Name name = ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).get());
        Phone phone = ParserUtil.parsePhone(argMultimap.getValue(PREFIX_PHONE).get());
        Email email = ParserUtil.parseEmail(argMultimap.getValue(PREFIX_EMAIL).get());
        Address address = ParserUtil.parseAddress(argMultimap.getValue(PREFIX_ADDRESS).get());
        Set<Tag> tagList = ParserUtil.parseTags(argMultimap.getAllValues(PREFIX_TAG));

        Person person = new Person(name, phone, email, address, tagList);

        return new AddCommand(person);
    }
```

**Review by kokerinks — CHANGES_REQUESTED** [review]
LGTM aside from the one comment, make sure to check codecov for un-tested lines too after the checkstyle errors are resolved

**Review by billyhoce — COMMENTED** [review]
Overall well written code and comprehensive test cases! Other than the few things mentioned, LGTM!

File: src/main/java/seedu/address/logic/commands/AddPointsCommand.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        this.points = points;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        List<Person> lastShownList = model.getFilteredPersonList();
+
+        Optional<Person> personOptional = lastShownList.stream()
+                .filter(person -> person.getName().equals(this.name))
+                .findFirst();
```
[the remark this answers is not in the document]
Comment (kinjalagarwal1810) [reply src/main/java/seedu/address/logic/commands/AddPointsCommand.java:53]: Yup okay makes sense!

**Review by kokerinks — APPROVED** [review]

**Pull request merged**